circus-train-s3-s3-copier/src/test/java/com/hotels/bdp/circustrain/s3s3copier/S3S3CopierTest.java

```diff
@@ -291,23 +291,23 @@ public void copyCheckTransferManagerIsShutdownWhenCopyExceptionsAreThrown() thro
     Path replicaLocation = new Path("s3://target/");
     List<Path> sourceSubLocations = new ArrayList<>();
 
-    RetryableTransferManagerFactory mockedRetryableTransferManagerFactory = Mockito.mock(RetryableTransferManagerFactory.class);
-    RetryableTransferManager mockedRetryableTransferManager = Mockito.mock(RetryableTransferManager.class);
-    when(mockedRetryableTransferManagerFactory.newInstance(any(AmazonS3.class), eq(s3S3CopierOptions), any(AmazonS3.class)))
-        .thenReturn(mockedRetryableTransferManager);
+    TransferManagerFactory mockedTransferManagerFactory = Mockito.mock(TransferManagerFactory.class);
+    TransferManager mockedTransferManager = Mockito.mock(TransferManager.class);
+    when(mockedTransferManagerFactory.newInstance(any(AmazonS3.class), eq(s3S3CopierOptions)))
+        .thenReturn(mockedTransferManager);
     Copy copy = Mockito.mock(Copy.class);
     when(copy.getProgress()).thenReturn(new TransferProgress());
-    when(mockedRetryableTransferManager.copy(any(CopyObjectRequest.class),
+    when(mockedTransferManager.copy(any(CopyObjectRequest.class), any(AmazonS3.class),
         any(TransferStateChangeListener.class))).thenReturn(copy);
     doThrow(new AmazonClientException("cause")).when(copy).waitForCompletion();
     S3S3Copier s3s3Copier = new S3S3Copier(sourceBaseLocation, sourceSubLocations, replicaLocation, s3ClientFactory,
-        mockedRetryableTransferManagerFactory, listObjectsRequestFactory, registry, s3S3CopierOptions);
+        mockedTransferManagerFactory, listObjectsRequestFactory, registry, s3S3CopierOptions);
     try {
       s3s3Copier.copy();
       fail("exception should have been thrown");
     } catch (CircusTrainException e) {
-      verify(mockedRetryableTransferManager).shutdownNow();
-      assertThat(e.getCause().getMessage(), is("cause"));
+      verify(mockedTransferManager).shutdownNow();
+      assertThat(e.getMessage(), is("1 job(s) failed the maximum number of copy attempts, 3"));
```

Comment:
We could check something like 

`verify(mockedTransferManager, Mockito.times(3)).copy(captor.capture(), any(AmazonS3.class), any(TransferStateChangeListener.class));`

here. Then possibly rename the method to be "maxRetriesExceeded".